src/App.tsx

```diff
@@ -14,6 +38,124 @@ export const App: React.FC = () => {
     }
   }, []);
 
+  const getTodosFromApi = async (userId: number) => {
+    try {
+      const response = await getTodos(userId);
+
+      setTodos(response);
+      setIsAdding(null);
+      setIsEditing(null);
+    } catch (error) {
+      throw new Error('Error on loading todos');
+    }
+  };
+
+  const createTodo = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!inputTitle) {
+      setInputError(true);
+      setIsHidden(false);
+
+      return;
+    }
+
+    if (user) {
+      const newTodo = {
+        userId: user.id,
+        title: inputTitle,
+        completed: false,
+      };
+
+      setIsAdding(newTodo);
+
+      try {
+        await createTodos(newTodo);
+      } catch {
+        setAddError(true);
+        setIsHidden(false);
+        setIsAdding(null);
+      }
+
+      getTodosFromApi(user.id);
+      setInputTitle('');
+    }
+  };
+
+  const selectCompleted = async (todo: Todo) => {
+    try {
+      await updateTodo(todo, !todo.completed);
+    } catch {
+      setUpdateError(true);
+      setIsHidden(false);
+    }
+
+    if (user) {
+      getTodosFromApi(user.id);
+    }
+  };
+
+  const selectAll = () => {
+    todos.map(async (todo) => {
+      if (!todo.completed && isAllSelected) {
+        await selectCompleted(todo);
+      } else if (todo.completed && !isAllSelected) {
+        await selectCompleted(todo);
+      }
+    });
+
+    setIsAllSelected(prev => !prev);
```

Comment:
should we change this value before our functions will update data on server? 

let's use Promise.all to await until our requests will be completed